docs/decisions/0013-backend-for-frontend-api.md

```diff
@@ -0,0 +1,69 @@
+# 0013. Introduction of the backend-for-frontend orchaestration API
+
+## Status
+
+Accepted (December 2024)
+
+## Context
+
+As part of supporting the feature of auto enrollment of default enrollment intentions ([see ADR](https://github.com/openedx/edx-enterprise/blob/master/docs/decisions/0015-default-enrollments.rst))
+and auto-apply of subscription plan licenses, it was determined the sequencing of API calls to resolve in the existing 
+frontend root loader would result in a new set of waterfall API calls on top of existing reconciliation of the current status of
+the user's subsidies when arriving on the enterprise learner portal. Furthermore, in support of including this feature,
+it required additional business logic to be parsed about the current state of the requester's association to the enterprise,
+the current state of their subsidies (and auto-apply of subscription licenses), and the eventual auto enrollment of a 
+default enrollment intention.
+
+## Decision
+
+The recognition of the additional maintenance of business logic with the default enrollment flow along with the overhead
+of resolving the series of API calls required to resolve the requirements of the feature resulted in the creation of a 
+backend-for-frontend (BFF) layer within the enterprise-access service ([see ADR](https://github.com/openedx/enterprise-access/blob/main/docs/decisions/)).
+
+A BFF layer is a accepted architectural consideration when designing for microservices when the consideration of consolidating
+business logic dependent on multiple services and an intentional reduction of latency between N number of services required to 
+determine the resultant business logic outcome versus a single API call with the logic already resolved to be used.
+
+For the enterprise learner portal implementation, currently, the dashboard route, `/:enterpriseSlug`, is the only route currently
+implementing the BFF layer on _specific_ API calls for the sole intention of supporting, auto apply licenses and auto enrollment via default intentions.
+
+### Making the BFF API call via `useBFF`
+
+At a high level, the implications to the frontend is the inclusion of the following hooks, services, and keys and constraints: 
+
+- `fetchEnterpriseLearnerDashboard` ([source](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/337a7b44d94d8be5d9233e80e2fa0e2de72d165c/src/components/app/data/services/bffs.ts))
+  - This is the function that makes the service call to the BFF. It requires either a `entepriseSlug` or `enterpriseUuid` to resolve successfully. 
+- Creation of the query key ([source](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/337a7b44d94d8be5d9233e80e2fa0e2de72d165c/src/components/app/data/queries/queryKeyFactory.js#L267))
```

Comment:
nit: it seems the pattern in these list items is to include the function name first, followed by the description. for example, the line item above starts with `fetchEnterpriseLearnerDashboard` and the line item below starts with `useBFF`. Should this line item start with `queryEnterpriseLearnerDashboardBFF`?